docs/source/language/snql.rst

```diff
@@ -1,5 +1,190 @@
+and ``RHS`` are literal values or expressions. ``OP`` refers to a specific
+operator to compare the two values. These operators are one of
+``=, !=, <, <=, >, >=, IN, NOT IN, LIKE, NOT LIKE, IS NULL, IS NOT NULL``.
+Note that the ``RHS`` is optional when using an operator like ``IS NULL``.
+
+Conditions can be combined using the boolean keywords ``AND`` or ``OR``.
+They can also be grouped using ``()``.
+
+Some conditions will be mandatory to provide a valid query depending on
+the entity. For example the Transactions entity requires a project id
+condition and a time range condition.
+
+HAVING
+======
+
+Works like the WHERE clause but it is applied after the aggregations declared
+in the SELECT clause. So we can apply conditions on the result of an aggregation
+function here.
+
+ORDER BY
+========
+
+Specify the expression(s) to order the result set on.
+
+LIMIT BY/LIMIT/OFFSET
+=====================
+
+Pretty self explanatory, they take integers and set the corresponding
+values in the Clickhouse query. If a query doesn't specify the limit or
+offset, they will be defaulted to 1000 and 0 respectively.
+
+GRANULARITY
+===========
+
+An integer representing the granularity to group time based results.
+
+TOTALS
+======
+
+If set to True, the response from Snuba will have a ``"totals"`` key that
+contains the total values across all the selected rows.
+
+SAMPLE
+======
+
+If a sampling rate isn't provided by a node in the ``MATCH`` clause, then it
+can be specified here. In this case, Snuba will assign the sample right to
+one of the nodes in the query. A sample can be either a float between 0 and
+1, representing a percentage of rows to sample.
+
+Or it can be an integer greater 1 which represents the number of rows to sample.
+
+JOINs
+=====
+
+A multi node subgraph is a subgraph that includes multiple relationships
+between different nodes. We only support 1..n, n..1 and 1..1 directed
+relationships between nodes. No m..n for now.::
+
+    MATCH (g: groups) -[contains]-> (e: events)
+
+
+The ``contains`` keyword represents the 1:n relationship defined in the
+data model between groups and events. The tuples provided to the where
+clause look exactly like the ones produced by a left join.::
+
+    [
+        {"g.id": 10, "g.title": "error", "e.id": "123123123123", ....}
+        {"g.id": 10, "g.title": "error", "e.id": "456456456456", ....}
+        {"g.id": 10, "g.title": "error", "e.id": "789789789789", ....}
+    ]
+
+To query a graph with multiple relationships in the MATCH clause, multiple
+two node relationships can be comma separated as long as the graph between
+all nodes referenced is connected. In the example below, ``groups`` is
+connected to all the relationships.::
+
+    MATCH (g: groups) -[contains]-> (e: events),
+                (g: groups) -[assigned]-> (a: groupAssignnee),
+                (g: groups) -[impacts]-> (r: groupReleases)
+    WHERE .....
+
+Sub-queries
```

Comment:
Should we have an "examples" section of the documentation for queries like this? This section doesn't seem to do much besides show the example.

Reply:
I added examples inline for sub queries and joins. We can add more later, but these should be enough to understand the basics.